File: by-kotlin/kt-boj/solutions/1978-소수 찾기.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import kotlin.math.sqrt

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val n = readLine().toInt()
    val numbers = readLine().split(" ").map { it.toInt() }
    var cnt = 0

    for (num in numbers) {
        if (isPrime(num)) cnt++
    }

    print(cnt)
}

fun isPrime(n: Int): Boolean {
    if (n == 1) return false

    val k = sqrt(n.toFloat()).toInt()
    for (i in 2..k) {
        if (n % i == 0) return false
    }
    return true
}

File: by-python/py-boj/1978-소수 찾기.py
def is_prime(n):
    if n == 1: return False

    k = int(n ** 0.5)
    for i in range(2, k + 1):
        if n % i == 0:
            return False
    else:
        return True


def solution(numbers):
    cnt = 0
    for n in numbers:
        if is_prime(n):
            cnt += 1
    return cnt


if __name__ == '__main__':
    n = int(input())
    numbers = list(map(int, input().split()))
    print(solution(numbers))
